gobblin-service/src/test/java/org/apache/gobblin/service/modules/core/GobblinServiceHATest.java

```diff
@@ -546,6 +551,30 @@ public void testKillNode() throws Exception {
 
     Assert.assertTrue(assertSuccess, "New master should take over all old master jobs.");
 
+    // Check eventbus was registered with new leader
+    attempt = 0;
+    assertSuccess = false;
+    while (attempt < 10) {
```

Comment:
AssertWithBackoff instead of while(..) {..sleep(..)}?

Reply:
How would you use AssertWithBackoff for a method that is throwing an exception instead of returning true/false? I guess only way is to wrap it in another method.